recipes/gf-complete/all/conanfile.py

```diff
@@ -1,162 +1,198 @@
-
-    @contextlib.contextmanager
-    def _build_context(self):
-        if self.settings.compiler == "Visual Studio":
-            with tools.vcvars(self):
-                env = {
-                    "CC": "{} cl -nologo".format(tools.unix_path(self.deps_user_info["automake"].compile)),
-                    "CXX": "{} cl -nologo".format(tools.unix_path(self.deps_user_info["automake"].compile)),
-                    "LD": "{} link -nologo".format(tools.unix_path(self.deps_user_info["automake"].compile)),
-                    "AR": "{} lib".format(tools.unix_path(self.deps_user_info["automake"].ar_lib)),
-                }
-                with tools.environment_append(env):
-                    yield
-        else:
-            yield
-
-    def _configure_autotools(self):
-        if self._autotools:
-            return self._autotools
-
-        self._autotools = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
-        self._autotools.libs = []
-        if self.settings.compiler == "Visual Studio":
-            self._autotools.flags.append("-FS")
-        elif "x86" in self.settings.arch:
-            self._autotools.flags.append("-mstackrealign")
-
+                replace_in_file(self, os.path.join(self.source_folder, subdir, "Makefile.am"), flag, "")
+
+    def generate(self):
+        env = VirtualBuildEnv(self)
+        env.generate()
+        if not cross_building(self):
+            env = VirtualRunEnv(self)
+            env.generate(scope="build")
+
+        tc = AutotoolsToolchain(self)
+        if not is_msvc(self) and "x86" in self.settings.arch:
+            tc.extra_cxxflags.append("-mstackrealign")
         yes_no = lambda v: "yes" if v else "no"
-        conf_args = [
-            "--enable-shared={}".format(yes_no(self.options.shared)),
-            "--enable-static={}".format(yes_no(not self.options.shared)),
-        ]
-
-        if "arm" in self.settings.arch:
+        if "arm" in str(self.settings.arch):
             if self.options.neon != "auto":
-                conf_args.append("--enable-neon={}".format(yes_no(self.options.neon)))
-
+                tc.configure_args.append("--enable-neon={}".format(yes_no(self.options.neon)))
         if self.settings.arch in ["x86", "x86_64"]:
             if self.options.sse != "auto":
-                conf_args.append("--enable-sse={}".format(yes_no(self.options.sse)))
-
+                tc.configure_args.append("--enable-sse={}".format(yes_no(self.options.sse)))
             if self.options.avx != "auto":
-                conf_args.append("--enable-avx={}".format(yes_no(self.options.avx)))
-
-        self._autotools.configure(args=conf_args, configure_dir=self._source_subfolder)
-
-        return self._autotools
+                tc.configure_args.append("--enable-avx={}".format(yes_no(self.options.avx)))
+        if is_msvc(self):
+            if self.options.shared:
+                tc.extra_ldflags.append("-no-undefined")
```

Comment:
Any insight into why this one is needed?

Reply:
It's been a while, but I think I added it as an Autotools substitute for `CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS`.
The `-no-undefined` was probably just to be safe.